Tests/scripts/utils/compare_staging_graph.py

```diff
@@ -140,6 +140,37 @@ def file_diff(output_path: Path, zip1_files: str, zip2_files: str, file: str, di
         print(f"could not diff files {file1_path}:{file2_path}: {e}")
 
 
+def compare(
+    output_path: Path,
+    marketplace: str,
+    zip_id_set: Path,
+    zip_graph: Path,
+    index_id_set_path: Path,
+    index_graph_path: Path,
+    collected_packs_id_set: Path,
+    collected_packs_graph: Path,
+    message: list[str],
+):
+    output_path.mkdir(exist_ok=True, parents=True)
+    # compare directories
+    dir_cmp = filecmp.dircmp(zip_id_set, zip_graph)
+    dir_cmp.report_full_closure()
+    message.append(f"Zip difference for {output_path}")
+    for file in dir_cmp.common_files:
+        pack = file.removesuffix(".zip")
+        if diff_files := compare_zips(zip_id_set / file, zip_graph / file, output_path / pack):
+            message.append(f'Detected differences in the following files for pack {pack}: {", ".join(diff_files)}')
+    if compare_indexes(index_id_set_path, index_graph_path, output_path):
+        message.append("Detected differences between index.json files")
```

Comment:
Can the differences be listed in the message?

Reply:
The difference could be too big for a slack message

Reply:
Can it be in an attachment?
Maybe in the log, with a link to it?

Reply:
Good enough. A (human) debugger could do a diff themselves...